scripts/devops_tasks/test_regression.py

```diff
@@ -0,0 +1,364 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# This script will run regression test for packages which are added as required package by other packages
+# Regression test ensures backword compatibility with released dependent package versions
+
+import argparse
+import glob
+import sys
+import os
+import logging
+from common_tasks import (
+    process_glob_string,
+    parse_setup,
+    run_check_call,
+    parse_require,
+    install_package_from_whl,
+    filter_dev_requirements,
+    OmmitType,
+)
+from git_helper import get_release_tag, checkout_code_repo, clone_repo
+
+AZURE_GLOB_STRING = "azure*"
+
+root_dir = os.path.abspath(os.path.join(os.path.abspath(__file__), "..", "..", ".."))
+test_tools_req_file = os.path.abspath(os.path.join(root_dir, "eng", "test_tools.txt"))
+
+GIT_REPO_NAME = "azure-sdk-for-python"
+GIT_MASTER_BRANCH = "master"
+VENV_NAME = "regressionenv"
+AZURE_SDK_FOR_PYTHON_GIT_URL = "https://github.com/Azure/azure-sdk-for-python.git"
+TEMP_FOLDER_NAME = ".tmp_code_path"
+COSMOS_TEST_ARG = "not cosmosEmulator"
```

Comment:
We should make this an argument in the build. Reason being, we may need to regression test emulator stuff and I'd rather not have to back out the hardcoding.

Reply:
Fine with it for debug, but before checkin should be moved I think.